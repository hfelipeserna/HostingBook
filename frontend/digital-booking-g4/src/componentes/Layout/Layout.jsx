import Header from "../Header/Header";
import Footer from "../Footer/Footer";
import Profile from "../Profile/Profile";

import style from "./Layout.module.css";
import SideNav from "../SideNav/SideNav";

export default function Layout(props){
    return(
        <>
            <Header>
                <Profile/>
            </Header>
            <main>
                <SideNav/>
                {props.children}
            </main>      
            <Footer></Footer>
        </>
    );
}