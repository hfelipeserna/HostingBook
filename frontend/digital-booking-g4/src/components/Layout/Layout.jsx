import Header from "../Header/Header";
import Footer from "../Footer/Footer";
import BloqueFechasDisponibles from "../BloqueFechasDisponiblesYReserva/BloqueFechasDisponibles";
import Searcher from "../Searcher/Searcher";

export default function Layout(props){
    return (
      <>
        <BloqueFechasDisponibles />
        <Searcher />
      </>
    );
}