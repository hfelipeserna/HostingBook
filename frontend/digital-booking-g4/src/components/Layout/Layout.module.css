main {
    margin-top: 76px;
    margin-bottom: 58px; 
    min-height: 100vh;
    display: flex;
    flex-direction: column;
}

@media screen and (max-width:768px) {
    main {
        margin-top: 102px;
    }
}