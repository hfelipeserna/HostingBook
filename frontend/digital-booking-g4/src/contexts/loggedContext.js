import { createContext } from "react";

const loggedContext = createContext();

export default loggedContext;