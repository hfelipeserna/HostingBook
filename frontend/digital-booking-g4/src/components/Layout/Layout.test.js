import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";
import Layout from "./Layout";
import loggedContext from "../../contexts/loggedContext";

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Profile tests", function () {
  let componente;

  beforeEach(() => {
    componente = render(
      <loggedContext.Provider value={{ isLogged: true }}>
        <Layout>Hola Mundo</Layout>
      </loggedContext.Provider>,
      { wrapper: MemoryRouter }
    );
  })

  test("Se renderiza el Layout con los children", () => {
    expect(componente.container).toHaveTextContent("Hola Mundo");
  });

  test("Devuelve el Layout correctamente", () => {
    expect(componente).toMatchSnapshot();
  });
});
