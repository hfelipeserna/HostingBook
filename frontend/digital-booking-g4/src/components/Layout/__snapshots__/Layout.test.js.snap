// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile tests Devuelve el Layout correctamente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="headerFijo"
      >
        <div
          class="contenedorHeader"
        >
          <a
            href="/"
          >
            <div
              class="headerIzquierda"
            >
              <img
                alt="logo"
                class="logoImg"
                src="logo1.png"
              />
              <div>
                <h1>
                  Hosting
                  <span>
                    Book
                  </span>
                </h1>
                <p>
                  Hospedamos tu próximo sueño
                </p>
              </div>
            </div>
          </a>
          <div
            class="menuHamburguesa"
          >
            <div
              class="menuHamburguesa"
            >
              <i
                class="bx bx-menu"
              />
            </div>
            <div />
            <div
              class="menuDrawer"
            >
              <div>
                <div
                  class="header"
                >
                  <span
                    class="cerrarSideNav"
                  >
                    X
                  </span>
                  <div
                    class="contenedorProfile"
                  >
                    <p
                      class="avatar"
                    >
                      BR
                    </p>
                    <div
                      class="contenedorNombre"
                    >
                      <p
                        class="saludo"
                      >
                        Hola,
                      </p>
                      <p
                        class="nombre"
                      >
                        Bruno Rodriguez
                      </p>
                    </div>
                    <span
                      class="cerrar"
                      data-testid="btnCerrarSesion"
                    >
                      X
                    </span>
                  </div>
                </div>
                <div
                  class="main"
                >
                  <p
                    class="cerrarSesion"
                  >
                    ¿Deseas
                    <span>
                       cerrar sesión
                    </span>
                    ?
                  </p>
                </div>
                <div
                  class="footer"
                >
                  <i
                    class="bx bxl-facebook-circle"
                  />
                  <i
                    class="bx bxl-linkedin"
                  />
                  <i
                    class="bx bxl-twitter"
                  />
                  <i
                    class="bx bxl-instagram-alt"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="headerDerecha"
          >
            <div
              class="contenedorProfile"
            >
              <p
                class="avatar"
              >
                BR
              </p>
              <div
                class="contenedorNombre"
              >
                <p
                  class="saludo"
                >
                  Hola,
                </p>
                <p
                  class="nombre"
                >
                  Bruno Rodriguez
                </p>
              </div>
              <span
                class="cerrar"
                data-testid="btnCerrarSesion"
              >
                X
              </span>
            </div>
          </div>
        </div>
      </header>
      <main>
        Hola Mundo
      </main>
      <footer
        class="footerFijo"
      >
        <div
          class="contenedorFooter"
        >
          <div
            class="footerIzquierda"
          >
            <p>
              ©2021 Hosting Book
            </p>
          </div>
          <div
            class="footerDerecha"
          >
            <i
              class="bx bxl-facebook-circle"
            />
            <i
              class="bx bxl-linkedin"
            />
            <i
              class="bx bxl-twitter"
            />
            <i
              class="bx bxl-instagram-alt"
            />
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <header
      class="headerFijo"
    >
      <div
        class="contenedorHeader"
      >
        <a
          href="/"
        >
          <div
            class="headerIzquierda"
          >
            <img
              alt="logo"
              class="logoImg"
              src="logo1.png"
            />
            <div>
              <h1>
                Hosting
                <span>
                  Book
                </span>
              </h1>
              <p>
                Hospedamos tu próximo sueño
              </p>
            </div>
          </div>
        </a>
        <div
          class="menuHamburguesa"
        >
          <div
            class="menuHamburguesa"
          >
            <i
              class="bx bx-menu"
            />
          </div>
          <div />
          <div
            class="menuDrawer"
          >
            <div>
              <div
                class="header"
              >
                <span
                  class="cerrarSideNav"
                >
                  X
                </span>
                <div
                  class="contenedorProfile"
                >
                  <p
                    class="avatar"
                  >
                    BR
                  </p>
                  <div
                    class="contenedorNombre"
                  >
                    <p
                      class="saludo"
                    >
                      Hola,
                    </p>
                    <p
                      class="nombre"
                    >
                      Bruno Rodriguez
                    </p>
                  </div>
                  <span
                    class="cerrar"
                    data-testid="btnCerrarSesion"
                  >
                    X
                  </span>
                </div>
              </div>
              <div
                class="main"
              >
                <p
                  class="cerrarSesion"
                >
                  ¿Deseas
                  <span>
                     cerrar sesión
                  </span>
                  ?
                </p>
              </div>
              <div
                class="footer"
              >
                <i
                  class="bx bxl-facebook-circle"
                />
                <i
                  class="bx bxl-linkedin"
                />
                <i
                  class="bx bxl-twitter"
                />
                <i
                  class="bx bxl-instagram-alt"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="headerDerecha"
        >
          <div
            class="contenedorProfile"
          >
            <p
              class="avatar"
            >
              BR
            </p>
            <div
              class="contenedorNombre"
            >
              <p
                class="saludo"
              >
                Hola,
              </p>
              <p
                class="nombre"
              >
                Bruno Rodriguez
              </p>
            </div>
            <span
              class="cerrar"
              data-testid="btnCerrarSesion"
            >
              X
            </span>
          </div>
        </div>
      </div>
    </header>
    <main>
      Hola Mundo
    </main>
    <footer
      class="footerFijo"
    >
      <div
        class="contenedorFooter"
      >
        <div
          class="footerIzquierda"
        >
          <p>
            ©2021 Hosting Book
          </p>
        </div>
        <div
          class="footerDerecha"
        >
          <i
            class="bx bxl-facebook-circle"
          />
          <i
            class="bx bxl-linkedin"
          />
          <i
            class="bx bxl-twitter"
          />
          <i
            class="bx bxl-instagram-alt"
          />
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
